dwi_ml/data/hdf5_creation.py

```diff
@@ -0,0 +1,241 @@
+        Group data created by concatenating all files, standardized.
+    group_affine: np.ndarray
+        Affine for the group.
+    """
+    # First file will define data dimension and affine
+    first_file = subj_input_path.joinpath(file_list[0])
+    logging.debug('    Loading {}'.format(first_file))
+    group_data, group_affine, group_header = \
+        _load_and_check_volume_to4d(first_file)
+
+    # Other files must fit
+    for data_name in file_list[1:]:
+        data_file = subj_input_path.joinpath(data_name)
+        logging.debug('    Loading {}'.format(data_file))
+        data, _, _ = _load_and_check_volume_to4d(data_file)
+        try:
+            group_data = np.append(group_data, data, axis=-1)
+        except ImportError:
+            raise ImportError('Data file {} could not be added to data group '
+                              '{}. Wrong dimensions?'.format(data_file, group))
+
+    # Save unnormalized data
+    if save_intermediate:
+        logging.debug('  Saving intermediate non standardized files.')
+        group_image = nib.Nifti1Image(group_data, group_affine)
+        output_fname = "{}_nonstandardized.nii.gz".format(group)
+        nib.save(group_image,
+                 str(subj_output_path.joinpath(output_fname)))
+
+    # Standardize data
+    logging.debug('  Standardizing data')
+    standardized_group_data = standardize_data(group_data, subj_mask_data)
+
+    # Save standardized data
+    if save_intermediate:
+        logging.debug('  Saving intermediate standardized files.')
+        standardized_img = nib.Nifti1Image(standardized_group_data,
+                                           group_affine)
+        output_fname = "{}_standardized.nii.gz".format(group)
+        nib.save(standardized_img,
+                 str(subj_output_path.joinpath(output_fname)))
+
+    return standardized_group_data, group_affine, group_header
+
+
+def process_streamlines(bundles_dir: Path, bundles,
+                        header: nib.Nifti1Header, step_size: float,
+                        space: Space):
+    """Load and process a group of bundles and merge all streamlines
+    together.
+
+    Parameters
+    ----------
+    bundles_dir : Path
+        Path to bundles folder.
+    bundles: List[str]
+        List of the bundles filenames to load. If none, all bundles will be
+        used.
+    header : nib.Nifti1Header
+        Reference used to load and send the streamlines in voxel space and to
+        create final merged SFT.
+    step_size: float
+        Step size to resample streamlines. If none, compress streamlines.
+    space: Space
+        Space to place the tractograms.
+
+    Returns
+    -------
+    output_tractogram : StatefulTractogram
+        All streamlines in voxel space.
+    output_lengths : List[float]
+        The euclidean length of each streamline
+    """
+    # Silencing SFT's logger if our logging is in DEBUG mode
+    set_sft_logger_level('WARNING')
```

Comment:
I don't think I invented that. I must have copied it somewhere. Is that really useful?

Reply:
Well I wouldn't mind it. SFT's tend to produce *a lot* of output when in verbose mode.

Reply:
Thanks!